presto-geospatial/src/test/java/com/facebook/presto/plugin/geospatial/TestGeometryJtsUtils.java

```diff
@@ -0,0 +1,107 @@
+ * limitations under the License.
+ */
+package com.facebook.presto.plugin.geospatial;
+
+import com.facebook.presto.geospatial.GeometryJtsUtils;
+import io.airlift.slice.Slice;
+import io.airlift.slice.Slices;
+import org.locationtech.jts.geom.Geometry;
+import org.locationtech.jts.io.ParseException;
+import org.locationtech.jts.io.WKTReader;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.plugin.geospatial.GeoFunctions.stGeometryFromText;
+import static com.facebook.presto.testing.assertions.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+public class TestGeometryJtsUtils
+{
+    private static final WKTReader WKT_READER = new WKTReader();
+
+    @Test
+    public void testPoint()
+            throws Exception
+    {
+        testGeometry("POINT EMPTY");
+        testGeometry("POINT (1 2)");
+    }
+
+    @Test
+    public void testMultiPoint()
+            throws Exception
+    {
+        testGeometry("MULTIPOINT EMPTY");
+        testGeometry("MULTIPOINT ((1 2), (3 4))");
+    }
+
+    @Test
+    public void testLineString()
+            throws Exception
+    {
+        testGeometry("LINESTRING EMPTY");
+        testGeometry("LINESTRING (1 2, 3 4, 5 6)");
+    }
+
+    @Test
+    public void testMultiLineString()
+            throws Exception
+    {
+        testGeometry("MULTILINESTRING EMPTY");
+        testGeometry("MULTILINESTRING ((1 2, 3 4, 5 6), (10.1 11.2, 12.3 13.4))");
+    }
+
+    @Test
+    public void testPolygon()
+            throws Exception
+    {
+        testGeometry("POLYGON EMPTY");
+        testGeometry("POLYGON ((0 0, 0 1, 1 1, 1 0, 0 0))");
+        testGeometry("POLYGON ((0 0, 0 1, 1 1, 1 0, 0 0), (0.1 0.1, 0.2 0.1, 0.2 0.2, 0.1 0.2, 0.1 0.1))");
+    }
+
+    @Test
+    public void testMultiPolygon()
+            throws Exception
+    {
+        testGeometry("MULTIPOLYGON EMPTY");
+        testGeometry("MULTIPOLYGON (((1 1, 1 3, 3 3, 3 1, 1 1)), ((2 4, 2 6, 6 6, 2 4)))");
+        testGeometry("MULTIPOLYGON (((1 1, 1 3, 3 3, 3 1, 1 1)), ((2 4, 2 6, 6 6, 2 4)), ((0 0, 0 1, 1 1, 1 0, 0 0), (0.1 0.1, 0.2 0.1, 0.2 0.2, 0.1 0.2, 0.1 0.1)))");
+    }
+
+    @Test
+    public void testGeometryCollection()
+            throws Exception
+    {
+        testGeometry("GEOMETRYCOLLECTION EMPTY");
+        testGeometry("GEOMETRYCOLLECTION (POINT (1 2), LINESTRING (0 0, 1 2, 3 4), POLYGON ((0 0, 0 1, 1 1, 1 0, 0 0)))");
+    }
+
+    private static void testGeometry(String wkt)
+            throws ParseException
+    {
+        Slice slice = stGeometryFromText(Slices.utf8Slice(wkt));
+        Geometry expected = WKT_READER.read(wkt);
+        Geometry actual = GeometryJtsUtils.deserialize(slice);
+
+        // ESRI shape serialization format doesn't contain enough information
+        // to distinguish between empty LineString and MultiLineString or
+        // empty Polygon and MultiPolygon.
+        if (expected.isEmpty()) {
```

Comment:
If expected is empty you check whether actual is empty too, but doesn't the `assertEquals(actual, expected)` below handle empty geometries?

Reply:
`assertEquals(actual, expected)` fails if both geometries are empty, but one is LineString and the other is a MultiLineString.